lib/resque/job.rb

```diff
@@ -157,7 +157,8 @@ def perform
         # Resque::DontPerform is raised.
         begin
           before_hooks.each do |hook|
-            job.send(hook, *job_args)
+            result = [job.send(hook, *job_args)]
+            job_args = result if hook.to_s['with_arg_mapping']
```

Comment:
Changing behavior via specialized hook naming comes as a surprise. If we want to transform job args, our API should support it directly & clearly.

How does this compare to prepending a module that defines `perform` and `super`s to the original implementation?

Reply:
Ah, I see your comment @ https://github.com/resque/resque/pull/1555#issuecomment-292103977. In this case, it'd be natural to offer an API to serialize/deserialize arguments.